application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,311 @@
+            return ZONED_TIME_MIN;
+        }
+
+        return getLocalDateTime(pubDate, dateTimeFormat);
+    }
+
+    /**
+     * Given a list of RSS feed items, this function checks if the dates are valid
+     * <p>
+     * This assumes that all items share the same date format (as is usual in an RSS feed response),
+     * therefore it only checks the first item's date for the final result.
+     */
+    private static boolean isValidDateFormat(List<Item> rssFeeds, RSSFeed feedConfig) {
+        try {
+            final Item firstRssFeed = rssFeeds.getFirst();
+            String firstRssFeedPubDate = firstRssFeed.getPubDate().orElse(null);
+
+            if (firstRssFeedPubDate == null) {
+                return false;
+            }
+
+            getLocalDateTime(firstRssFeedPubDate, feedConfig.dateFormatterPattern());
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to find a text channel from a given RSS feed configuration.
+     */
+    private Optional<TextChannel> getTextChannelFromFeed(JDA jda, RSSFeed feed) {
+        // Attempt to find the target channel
+        if (feed.targetChannelPattern() != null) {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> targetChannelPatterns.get(feed).test(channel.getName()))
+                .findFirst();
+        }
+
+        // If the target channel was not found, use the fallback
+        return jda.getTextChannelCache()
+            .stream()
+            .filter(channel -> defaultChannelPattern.test(channel.getName()))
+            .findFirst();
+    }
+
+    /**
+     * Provides the {@link EmbedBuilder} from an RSS item used for sending RSS posts.
+     */
+    private static EmbedBuilder constructEmbedMessage(Item item, RSSFeed feedConfig) {
+        final EmbedBuilder embedBuilder = new EmbedBuilder();
+        String title = item.getTitle().orElse("No title");
+        String titleLink = item.getLink().orElse("");
+        Optional<String> rawDescription = item.getDescription();
+
+        // Set the item's timestamp to the embed if found
+        item.getPubDate()
+            .ifPresent(date -> embedBuilder
+                .setTimestamp(getLocalDateTime(date, feedConfig.dateFormatterPattern())));
+
+        embedBuilder.setTitle(title, titleLink);
+        embedBuilder.setAuthor(item.getChannel().getLink());
+
+        // Process embed's description if a raw description was provided
+        if (rawDescription.isPresent() && !rawDescription.get().isEmpty()) {
+            Document fullDescription =
+                    Jsoup.parse(StringEscapeUtils.unescapeHtml4(rawDescription.get()));
+            StringBuilder finalDescription = new StringBuilder();
+            fullDescription.body()
+                .select("*")
+                .forEach(p -> finalDescription.append(p.text()).append(". "));
```

Comment:
Use a Stream instead of a StringBuilder (`Document#stream()`)

Reply:
That is done! ✅